cookbook/asset/custom_version_strategy.rst

```diff
@@ -0,0 +1,263 @@
+            "js/script.js": "f9c7afd05729f10f55b689f36bb20172",
+            "css/style.css": "91cd067f79a5839536b46c494c4272d8"
+        }
+
+Create Compiler Pass
+~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    // src/AppBundle/DependencyInjection/Compiler/OverrideAssetsDefaultPackagePass.php
+    namespace AppBundle\DependencyInjection\Compiler;
+
+    use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+    use Symfony\Component\DependencyInjection\ContainerBuilder;
+    use Symfony\Component\DependencyInjection\Reference;
+
+    class OverrideAssetsDefaultPackagePass implements CompilerPassInterface
+    {
+        public function process(ContainerBuilder $container)
+        {
+            $definition = $container->getDefinition('assets._default_package');
+            $definition->replaceArgument(1, new Reference('app.assets.buster_version_strategy'));
+        }
+    }
+
+The code above fetches the service definition of the default package, and replaces
+its second argument (the version strategy).
+
+Register Compiler Pass
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: php
+
+    // src/AppBundle/AppBundle.php
+    namespace AppBundle;
+
+    use AppBundle\DependencyInjection\Compiler\OverrideAssetsDefaultPackagePass;
+    use Symfony\Component\DependencyInjection\ContainerBuilder;
+    use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+    class AppBundle extends Bundle
+    {
+        public function build(ContainerBuilder $container)
+        {
+            parent::build($container);
+
+            // only register in prod environment
+            if ('prod' === $container->getParameter('kernel.environment')) {
+                $container->addCompilerPass(new OverrideAssetsDefaultPackagePass());
+            }
+        }
+    }
+
+See :doc:`/cookbook/service_container/compiler_passes` for more information
+on how to use compiler passes.
+
+Register Services
+~~~~~~~~~~~~~~~~~
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.assets.buster_version_strategy:
+                class: AppBundle\Asset\VersionStrategy\BusterVersionStrategy
+                arguments:
+                    - manifest_path: "%kernel.root_dir%/../busters.json"
+                      format: "%%s?version=%%s"
+                public: false
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd"
+        >
+            <services>
+                <service id="app.assets.buster_version_strategy" class="AppBundle\Asset\VersionStrategy\BusterVersionStrategy" public="false">
+                    <argument type="collection">
```

Comment:
hmm no sorry. It does match them. I looked at it too fast. However, I don't think passing an array of option is worth it. Just configure things with normal arguments.